exercises/robot-simulator/config/checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,369 @@
+        </module>
+
+        <module name="ImportOrder">
+            <!-- Checks for out of order import statements. -->
+            <property name="severity" value="error" />
+            <property name="ordered" value="true" />
+            <property name="groups" value="/([^j]|.[^a]|..[^v]|...[^a])/,/^javax?\./" />
+            <!-- This ensures that static imports go to the end. -->
+            <property name="option" value="bottom" />
+            <property name="tokens" value="STATIC_IMPORT, IMPORT" />
+        </module>
+
+        <module name="IllegalImport">
+            <property name="illegalPkgs" value="junit.framework" />
+        </module>
+
+        <module name="UnusedImports" />
+
+        <!--
+
+        METHOD LENGTH CHECKS
+
+        -->
+
+        <module name="MethodLength">
+            <property name="tokens" value="METHOD_DEF" />
+            <property name="max" value="300" />
+            <property name="countEmpty" value="false" />
+            <property name="severity" value="warning" />
+        </module>
+
+        <!--
+
+        NAMING CHECKS
+
+        -->
+
+        <!-- Item 38 - Adhere to generally accepted naming conventions -->
+
+        <module name="PackageName">
+            <!-- Validates identifiers for package names against the
+              supplied expression. -->
+            <!-- Here the default checkstyle rule restricts package name parts to
+              seven characters, this is not in line with common practice at Google.
+            -->
+            <property name="format" value="^[a-z]+(\.[a-z][a-z0-9]{1,})*$" />
+            <property name="severity" value="error" />
+        </module>
+
+        <module name="TypeNameCheck">
+            <!-- Validates static, final fields against the
+            expression "^[A-Z][a-zA-Z0-9]*$". -->
+            <metadata name="altname" value="TypeName" />
+            <property name="severity" value="error" />
+        </module>
+
+        <module name="ConstantNameCheck">
+            <!-- Validates non-private, static, final fields against the supplied
+            public/package final fields "^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$". -->
+            <metadata name="altname" value="ConstantName" />
+            <property name="applyToPublic" value="true" />
+            <property name="applyToProtected" value="true" />
+            <property name="applyToPackage" value="true" />
+            <property name="applyToPrivate" value="false" />
+            <property name="format" value="^([A-Z][A-Z0-9]*(_[A-Z0-9]+)*|FLAG_.*)$" />
+            <message key="name.invalidPattern"
+                value="Variable ''{0}'' should be in ALL_CAPS (if it is a constant) or be private (otherwise)." />
+            <property name="severity" value="error" />
+        </module>
+
+        <module name="StaticVariableNameCheck">
+            <!-- Validates static, non-final fields against the supplied
+            expression "^[a-z][a-zA-Z0-9]*_?$". -->
```

Comment:
Why the underscore at the end? I wouldn't expect static variable names to end in an underscore?

Reply:
me too